docs/polaris/data-science-workflows/julia.md

```diff
@@ -0,0 +1,208 @@
+
+```bash
+export /gpfs/alpine/scratch/$USER/$PROJECT/julia_depot
+```
+
+## Programming Julia on Polaris
+There are three key components to using Julia for large-scale computations:
+
+1. MPI support through [MPI.jl](https://github.com/JuliaParallel/MPI.jl)
+2. GPU support through [CUDA.jl](https://github.com/JuliaGPU/CUDA.jl)
+3. HDF5 support through [HDF5.jl](https://juliaio.github.io/HDF5.jl/stable/)
+
+In addition we recommend VSCode with the [Julia
+extension](https://www.julia-vscode.org/) for a modern IDE experience, together
+with the ssh-remote extension for remote interactive development.
+### MPI Support
+
+MPI support is provided through the [MPI.jl](https://github.com/JuliaParallel/MPI.jl).
+```julia
+julia> ] add MPI
+```
+This will install the MPI.jl package and default MPI prebuilt binaries provided by an artifact. For on-node debugging purposes the default artifact is sufficient. However, for large-scale computations, it is to use the system MPI library that is loaded via `module`. As of MPI.jl v0.20 this is handled through [MPIPrefences.jl](https://juliaparallel.org/MPI.jl/stable/configuration/#using_system_mpi).
+```
+julia --project -e 'using MPIPreferences; MPIPreferences.use_system_binary()'
+```
+### GPU Support
+
+NVIDIA GPU support is provided through the [CUDA.jl](https://github.com/JuliaGPU/CUDA.jl) package.
+```julia
+julia> ] add CUDA
+```
+In case you want write portable GPU kernels we highly recommend the [KernelAbstractions.jl](https://github.com/JuliaGPU/KernelAbstractions.jl) package. It provides a high-level abstraction for writing GPU kernels that can be compiled for different GPU backends.
+```julia
+julia> ] add KernelAbstractions
+```
+By loading either [oneAPI.jl](https://github.com/JuliaGPU/oneAPI.jl), [AMDGPU.jl](https://github.com/JuliaGPU/AMDGPU.jl), or [CUDA.jl](https://github.com/JuliaGPU/CUDA.jl) (see quickstart guide below).
+
+### HDF5 Support
+Parallel HDF5 support is provided by
+```
+module load cray-hdf5-parallel
+```
+After setting `export JULIA_HDF5_PATH=$HDF5_DIR` we can install the [HDF5.jl](https://juliaio.github.io/HDF5.jl/stable/) package.
+```julia
+julia> ] add HDF5
+```
+
+## Quickstart Guide
+The following example shows how to use MPI.jl, CUDA.jl, and HDF5.jl to write a
+parallel program that computes the sum of two vectors on the GPU and writes the
+result to an HDF5 file. A repository with the example code computing an
+approximation of pi can be found at
+[Polaris.jl](https://github.com/exanauts/Polaris.jl). In this repository, you will also find
+a `setup_polaris.sh` script that will build the HDF5.jl and MPI.jl package for the system libraries.
+The dependencies can be installed with the following commands:
+```bash
+julia --project
+```julia
```

Comment:
This part is rendered weird in GitHub Flavored Markdown for me--- does it look normal after building with MkDocs? Seems like it is missing closing ``` for each of the 3x code blocks

Reply:
Fixed.